packages/loot-core/src/server/budget/types/templates.d.ts

```diff
@@ -0,0 +1,82 @@
+interface BaseTemplate {
+  type: string;
+  priority?: number;
+  directive: string;
+}
+
+interface PercentageTemplate extends BaseTemplate {
+  type: 'percentage';
+  percent: number;
+  previous: boolean;
+  category: string;
+}
+
+interface WeekTemplate extends BaseTemplate {
+  type: 'week';
+  amount: number;
+  weeks: number | null;
+  starting: string;
+  limit?: { amount: number; hold: boolean };
+}
+
+interface ByTemplate extends BaseTemplate {
+  type: 'by';
+  amount: number;
+  month: string;
+  repeat?: { annual: boolean; repeat?: number };
+  from?: string;
+}
+
+interface SpendTemplate extends BaseTemplate {
+  type: 'spend';
+  amount: number;
+  month: string;
+  from: string;
+  repeat?: { annual: boolean; repeat?: number };
+}
+
+interface SimpleTemplate extends BaseTemplate {
+  type: 'simple';
+  monthly?: number;
+  limit?: { amount: number; hold: boolean };
+}
+
+interface ScheduleTemplate extends BaseTemplate {
+  type: 'schedule';
+  name: string;
+  full?: boolean;
+}
+
+interface RemainderTemplate extends BaseTemplate {
+  type: 'remainder';
+  weight: number;
+  limit?: { amount: number; hold: boolean };
```

Comment:
Do remainders pass along a limit? or is this for later work?

Reply:
I used the `goal-template.pegjs` as a source of truth for the types, so it may not actually being used (yet?) I can remove it if it's clearer

Reply:
I guess if its in there, then leave it.  I don't think that it gets used in the processing function though.